### ISSUE TITLE
Feature request: filter by multiple different substrings

### ISSUE THREAD

**Issue by coekuss** [JerwuQu/wlines#2]
I would like to use wlines for selecting web bookmarks from a text file, with different entries on each line tagged with keywords. I think it would be cool if it worked like this:

a search for `music art` matches entries including
both substring `music` _and_ substring `art`
instead of 
substring `music[space]art`

This is how I would use rofi on linux and it made it very comfortable to use. Great work on this software so far by the way, love the simplicity

**Comment by JerwuQu** [comment]
Hey!

Glad to see you're finding use for this project! I'm having trouble finding motivation for side-projects whilst I'm a full-time employee, but finding out I actually have users is a strong motivator :)

This sounds like a useful feature so I'll make sure it gets implemented sometime in future. I'd like to improve on the search feature-set a bit in general and this might be a good end to start in.

Currently the search doesn't reorder results at all but instead keeps the input-order and solely _filters_ based on the search.
Would you expect this feature to also reorder the results based on some metric? If so, could you try to explain how you would like it to work?

**Comment by coekuss** [comment]
Thanks for the reply, and I totally understand the side project difficulty. I wouldn't expect the results to need reordering, just a filter based on multiple criteria. A better explanation of my workflow would be I search the word `music` to filter my bookmarks by those that include keyword "music", then I want to _add_ the word `art` to the search (now `music art`), to then filter the results that include `music` by those that also include `art`, i.e. music websites that are also art websites.

**Comment by coekuss** [comment]
Probably best to also provide a real world example. Say I have these lines:

```
Soundcloud - art music
Spotify - music
Youtube - art content
Bandcamp - music art
Arena - art
```

Searching `music` would yield:
```
Soundcloud - art music
Spotify - music
Bandcamp - music art
```

Then searching `music art` would yield:
```
Soundcloud - art music
Bandcamp - music art
```

Then I can select the Bandcamp option by searching `music art band`:
```
Bandcamp - music art
```

Sort of like applying a new filter to the already-filtered results

**Comment by JerwuQu** [comment]
I went ahead and implemented this in b6c792e. The build can be found [here](https://github.com/JerwuQu/wlines/releases/tag/ci-b6c792e1f36b963c55c9ce9577243cd8019b3944).
You can use it by supplying `-fm keywords` to the CLI. 

Please give it a try and see if it works like you'd expect :)

**Comment by coekuss** [comment]
Working just as I hoped! Really appreciate the quick work, this is epic👍
